ui/perfherder/alerts/AlertsView.jsx

```diff
@@ -55,43 +55,30 @@ class AlertsView extends React.Component {
   }
 
   componentDidUpdate(prevProps, prevState) {
-    const { count, frameworkOptions } = this.state;
-    const { validated } = this.props;
-    const prevValitated = prevProps.validated;
+    const { count } = this.state;
 
     if (prevState.count !== count) {
       this.setState({ totalPages: this.generatePages(count) });
     }
 
-    // filters updated directly in the url
-    if (
-      validated.hideAssignedToOthers !== prevValitated.hideAssignedToOthers ||
-      validated.hideImprovements !== prevValitated.hideImprovements ||
-      validated.hideDwnToInv !== prevValitated.hideDwnToInv ||
-      validated.filterText !== prevValitated.filterText ||
-      validated.status !== prevValitated.status ||
-      validated.framework !== prevValitated.framework
-    ) {
-      this.setFiltersState(
-        this.getFiltersFromParams(validated, frameworkOptions),
-        false,
-      );
-    }
-
     const params = parseQueryParams(this.props.location.search);
+    const prevParams = parseQueryParams(prevProps.location.search);
     // we're using local state for id instead of validated.id because once
     // the user navigates from the id=<alert> view back to the main alerts view
     // the Validation component won't reset the id (since the query param doesn't exist
     // unless there is a value)
-    if (this.props.location.search !== prevProps.location.search) {
-      this.setState({ id: params.id || null }, this.fetchAlertSummaries);
-      if (params.id) {
-        validated.updateParams({ hideDwnToInv: 0 });
-      }
+    if (params.id !== prevParams.id) {
```

Comment:
Changing this to only be hit for selecting an alert, and navigating back, solved the issue with the pagination, because as soon as the page query param was updated in `PaginationGroup`, it would get caught here and the default page value of 1 in `fetchAlertSummaries` would override the previous changes.